### PR TITLE
add guidance regarding resource request/limit sizing

### PR DESCRIPTION
**Pull request by tzneal**
*Issue #, if available:*

N/A

*Description of changes:*

Add content regarding sizing of resource requests/limits.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: content/reliability/docs/dataplane.md [hunk 1]
```diff
@@ -163,6 +163,20 @@ For critical applications, consider defining `requests`=`limits` for the contain
...
 
+### Configure and Size Resource Requests/Limits for all Workloads
+
+Some general guidance can be applied to sizing resource requests and limits for workloads:
+
+- Do not specify resource limits on CPU.  In the absence of limits, the request acts as a weight on [how much relative CPU time containers get](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#how-pods-with-resource-limits-are-run). This allows your workloads to use the full CPU without an artificial limit or starvation.
+
+- For non-CPU resources, configuring `requests`=`limits` provides the most predictable behavior. If `requests`!=`limits`, the container also has its [QOS](https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/#qos-classes) reduced from Guaranteed to Burstable making it more likely to be evicted in the event of [node pressure](https://kubernetes.io/docs/concepts/scheduling-eviction/node-pressure-eviction/).
+
+- For non-CPU resources, do not specify a limit that is much larger than request.  The larger `limits` are configured relative to `requests`, the more likely nodes will be overcommitted leading to high chances of workload interruption.
```
Comment (bwagner5) [inline content/reliability/docs/dataplane.md:174]:
```suggestion
- For non-CPU resources, do not specify a limit that is much larger than the request.  The larger `limits` are configured relative to `requests`, the more likely nodes will be overcommitted leading to high chances of workload interruption.
- ```

File: content/reliability/docs/dataplane.md [hunk 2]
```diff
@@ -163,6 +163,20 @@ For critical applications, consider defining `requests`=`limits` for the contain
...
+
+- Correctly sized requests are particularly important when using an autos-scaling solution like [Karpenter](/karpenter/) or [Cluster AutoScaler](/cluster-autoscaling).  These tools look at your workload requests to determine the number and size of nodes to be provisioned. If your requests are too small with larger limits, you may find your workloads being evicted or OOM killed if they have been tightly packed on a node.
```
Comment (bwagner5) [inline content/reliability/docs/dataplane.md:176]:
```suggestion
- Correctly sized requests are particularly important when using a node auto-scaling solution like [Karpenter](/karpenter/) or [Cluster AutoScaler](/cluster-autoscaling).  These tools look at your workload requests to determine the number and size of nodes to be provisioned. If your requests are too small, with larger limits, you may find your workloads evicted or OOM killed if they have been tightly packed on a node.
```

File: content/reliability/docs/dataplane.md [hunk 3]
```diff
@@ -163,6 +163,20 @@ For critical applications, consider defining `requests`=`limits` for the contain
...
+
+Determining resource requests can be difficult, but tools like the [Vertical Pod Autoscaler](https://github.com/kubernetes/autoscaler/tree/master/vertical-pod-autoscaler) can help you 'right-size' the requests by observing container resource usage at runtime.
```
Comment (bwagner5) [inline content/reliability/docs/dataplane.md:178]: might be useful to link this? https://mhausenblas.info/right-size-guide/

Comment (tzneal) [reply content/reliability/docs/dataplane.md:178]: I may leave it off for now, VPA seems to be the standard way to do this, but RSG does look neat.

**Review by bwagner5 — APPROVED** [review]
lgtm! nice!

File: content/reliability/docs/dataplane.md (hunk 3 above)
Comment (jicowan) [reply content/reliability/docs/dataplane.md:178]: Continuous profiling with a tool such a [Prodfiler](https://prodfiler.com/) or [Parca](https://www.parca.dev/) is better than RSG. I'm fine with mentioning these options or [Goldilocks](https://github.com/FairwindsOps/goldilocks) which is a nice UI for VPA.

Comment (jicowan) [reply content/reliability/docs/dataplane.md:178]: @tzneal Are you going to link to this updated guidance in the Karpenter guidance?

Comment (tzneal) [reply content/reliability/docs/dataplane.md]: Good call, I added a link back and just listed all of the tools mentioned. Happy to rework the wording if there are any suggestions.

**Review by jicowan — APPROVED** [review]
Thanks for the submission!

**Pull request merged**